### DISCUSSION TITLE
Errors in production deployment not happening in preview deployment

### DISCUSSION THREAD

**Discussion by rubybrowncoat** [vercel/community#707] [Help] [answered]
I'm having a lot of trouble trying to track down hydration errors (418 nine times and 423 one time) that are only happening on a production deployment through Vercel (preview deployments, local developments and local builds do not present the errors at all, and no warnings).

These errors are a big problem because they are crashing the application when it is viewed in some in-app safari browsers (particularly for my use-case telegram) with a "Application error: a client-side exception has occurred" message. 

Is there something I can do to narrow down what is going on? Keeping in mind that I can't see the errors in preview or local environments.

EDIT:
I have deployed the exact same application to another Vercel project and I'm not seeing the errors in the new production deployment. Afterwards I have also moved all local storage and cookies, and I'm still not seeing the errors. What is going on? What could be causing this discrepancy?

**Comment by amyegan** [comment, marked as answer]
Hi @rubybrowncoat. There are some things you can try to help us narrow down what's going wrong. It sounds like you might have already tried some of these things, but I want to be certain.
- What are the main frameworks/tools you're using and their version numbers (i.e. Next 12.1.4, React 18...)
- Assuming Next.js, does it work if you run `npm run build` and then `npm run start` locally instead of `npm run dev`? 
- Does it fail in all browsers, or only Safari? Does it still fail after you clear your browser cache?
- Can you share links to the failing and working project builds?
- Can you identify any differences in settings between your original deployment Project and your new Project?
- If possible, sharing the your repo could help give a better picture of what's happening.

**Reply by rubybrowncoat** [reply]
- Next.js 12.2.0 is the main thing, doing UI with Chakra UI, and a slew of other little packages. Keep in mind that it's been happening since the beginning (a couple months now), across Next.js versions, across Chakra UI version, and across package additions and removals.
- If I do a local build myself, I don't see the error when I start it up locally
- I see there errors in all browsers I've tried, it just crashes in-app safari (not safari the app, just some in-app instances, Telegram and Facebook come to mind) 
- Here's the project where I'm seeing the errors: https://redattolo.vercel.app (You'd have to get past the introduction/tutorial by pressing the "Salta l'introduzione" button, as the tutorial page doesn't have the errors, which is weird in itself). And here's the same application deployed on a different project where I'm not seeing the errors at all: https://redattolo-bug-hunt.vercel.app
- I have made the two project purposefully identical, deployed the second project manually from the same codebase

I'm currently trying to isolate components to client-side only one at a time to see if they affect the errors, but I'm mindful of disruptions since the error appears only in the production deployment. It could very well be a misbehaving component somewhere but it's strange the errors are isolated to that one deployment in production and I've never seen them or warnings in development or in preview deployments.

**Reply by rubybrowncoat** [reply]
Just a heads up I've deployed a new version to production that doesn't exhibit the errors (I have made the header component dynamic and client only, will have to dig deeper into it to fix it) so the errors seem to be no more. 

I am back to square one as far as in-app browsers crashing goes because they keep crashing even now that I have squashed the errors.

**Reply by amyegan** [reply]
This is a strange issue. ~~I'm going to keep looking into it, but wanted to let you know that the original site is working for me in Facebook in-app browser without any "Application error: a client-side exception has occurred" message.~~ 

Edit: Looks like you fixed it before I was able to test, so that explains why I wasn't seeing the error. I'm glad you have a workaround for that issue at least.

**Reply by rubybrowncoat** [reply]
Wondering if you have any suggestions about debugging these in-app browser errors happening in a production deployment. (Because I'm still getting them, and now that you are mentioning not seeing the error while I'm still getting it I'm wondering if it's a data-specific issue with my instance be it cookies or storage, or a version issue)

**Reply by amyegan** [reply]
My best guess for what's going on is something with cookies/storage/cache or possibly an OS or app version issue, as you suggest.
